.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,41 @@
+name: unit-tests
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  unit-tests:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: linux
+            runs-on: ubuntu-22.04
+            arch: x86_64
+
+    name: "${{ matrix.os }}-${{ matrix.arch }}"
+    runs-on: ${{ matrix.runs-on }}
+
```

Comment:
please add `concurrency` which will cancel previous commit runs if new commits are pushed on the pr

https://github.com/crashappsec/chalk/blob/b583d27c6648b6860149b3b3550a29185c9015ed/.github/workflows/tests.yml#L53-L58

Reply:
I feel like its still wasting some minutes somewhere but fair enough. but I strongly feel it should be consistent across the repo so if we dont want that behavior, lets match it everywhere